### PR TITLE
HAAR-549: Allow Non-Alpha chars('’-) in first and last name

### PR DESCRIPTION
**Pull request by ravmoj**
Allow Non-Alpha chars('’-) in first and last name

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/resource/UserManagementResourceIntTest.kt [hunk 1]
```diff
@@ -436,6 +436,44 @@ class UserManagementResourceIntTest : IntegrationTestBase() {
         .jsonPath("lastName").isEqualTo("O'neillastname")
     }
 
+    @Test
+    fun `can change name of a user that includes an hyphen in fist and lastname  name`() {
```
Comment (karenmillermoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/resource/UserManagementResourceIntTest.kt:440]: fiRst typo

File: src/test/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/resource/UserManagementResourceIntTest.kt [hunk 2]
```diff
@@ -436,6 +436,44 @@ class UserManagementResourceIntTest : IntegrationTestBase() {
...
+        )
+        .exchange()
+        .expectStatus().isOk
+        .expectBody()
+        .jsonPath("firstName").isEqualTo("Sarah-louise")
+        .jsonPath("lastName").isEqualTo("O'neillastname")
+    }
+
+    @Test
+    fun `can change name of a user that includes an forward apostrophe in lastname  name`() {
```
Comment (karenmillermoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/resource/UserManagementResourceIntTest.kt:459]: can not change name of a user that includes an forward apostrophe in last name`

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/resource/UserManagementResource.kt [hunk 3]
```diff
@@ -312,13 +312,13 @@ class UserManagementResource(
...
-    message = "First name must consist of alphabetical characters only and a max 35 chars"
+    regexp = "^[A-Za-z'-]{1,35}$",
+    message = "First name must consist of alphabetical characters, hyphen or an apostrophe only and a max 35 chars"
   ) val firstName: String,
   @Schema(description = "Last name of the user", example = "Smith", required = true)
   @field:Pattern(
-    regexp = "^[A-Za-z']{1,35}$",
-    message = "Last name must consist of alphabetical characters or an apostrophe only and a max 35 chars"
+    regexp = "^[A-Za-z'-]{1,35}$",
+    message = "Last name must consist of alphabetical characters, hyphen or an apostrophe only and a max 35 chars"
```
Comment (karenmillermoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/resource/UserManagementResource.kt:321]: Should really say "a hyphen" (same on line 316)

File: src/test/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/resource/UserManagementResourceIntTest.kt (hunk 1 above)
Comment (ravmoj) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/resource/UserManagementResourceIntTest.kt]: Updated

**Review by karenmillermoj — APPROVED** [review]
lgtm

**Pull request merged**